packages/backend-common/src/service/lib/config.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ConfigReader } from '@backstage/config';
+import { CorsOptions } from 'cors';
+
+export type BaseOptions = {
+  listenPort?: number;
+  listenHost?: string;
+};
+
+/**
+ * Reads some base options out of a config object.
+ *
+ * @param config The root of a backend config object
+ * @returns A base options object
+ *
+ * @example
+ * ```json
+ * {
+ *   baseUrl: "http://localhost:7000",
+ *   listen: "0.0.0.0:7000"
+ * }
+ * ```
+ */
+export function readBaseOptions(config: ConfigReader): BaseOptions {
+  // TODO(freben): Expand this to support more addresses and perhaps optional
+  const { host, port } = parseListenAddress(config.getString('listen'));
+  return removeUnknown({
+    listenPort: port,
+    listenHost: host,
+  });
+}
+
+/**
+ * Attempts to read a CORS options object from the root of a config object.
+ *
+ * @param config The root of a backend config object
+ * @returns A CORS options object, or undefined if not specified
+ *
+ * @example
+ * ```json
+ * {
+ *   cors: {
+ *    origin: "http://localhost:3000",
+ *    credentials: true
+ *   }
+ * }
+ * ```
+ */
+export function readCorsOptions(config: ConfigReader): CorsOptions | undefined {
+  const cc = config.getOptionalConfig('cors');
+  if (!cc) {
+    return undefined;
+  }
+
+  return removeUnknown({
+    origin: getOptionalStringOrStrings(cc, 'origin'),
+    methods: getOptionalStringOrStrings(cc, 'methods'),
```

Comment:
Not sure methods should be configurable tbh? seems like an implementation concern.

Reply:
It's just one of the options that the cors thing accepts. If we make this more advanced in the future and wire things up more tightly, we could make it deduce the methods from the list of endpoints actually exposed. But it would take more time than I'm willing to spend right now